eth/consensus/pow.py

```diff
@@ -30,23 +31,33 @@
 
 
 # Type annotation here is to ensure we don't accidentally use strings instead of bytes.
-cache_seeds = [b'\x00' * 32]  # type: List[bytes]
+cache_seeds = {0: b'\x00' * 32}  # type: Dict[int, bytes]
```

Comment:
Probably not worth it unless it turns out to be easy, but it appears that `cache_by_seeds` naturally regulates it's maximum size.  Is there a way we can use a `weakref.WeakValueDict` to have this automatically clean itself out based on what is stored in the `cache_by_seed` data structure.

Given these are just 32 byte values and epochs are long, not a big deal at all.

Reply:
Hm, not sure where the strong reference would come from. Also, in case we want to go verify an old seed, we might trigger a regen of seeds from 0. It opens a lot of questions, and I don't *think* the upside is big enough to justify it.